### PR TITLE
refactor and added specs for OverviewDetailsKnativeResourcesTab

### PR DESCRIPTION
**Pull request by invincibleJai**
refactor and added specs for OverviewDetailsKnativeResourcesTab

Tracks : https://issues.redhat.com/browse/ODC-2557

### CODE REVIEW

File: frontend/packages/knative-plugin/src/components/overview/__tests__/OverviewDetailsKnativeResourcesTab.spec.tsx [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+import { OverviewItem } from '@console/shared';
+import { MockKnativeResources } from '@console/dev-console/src/components/topology/__tests__/topology-knative-test-data';
+import OverviewDetailsKnativeResourcesTab from '../OverviewDetailsKnativeResourcesTab';
+import KnativeServiceResources from '../KnativeServiceResources';
+import KnativeRevisionResources from '../KnativeRevisionResources';
+import EventSinkServicesOverviewList from '../EventSinkServicesOverviewList';
+
+describe('OverviewDetailsKnativeResourcesTab', () => {
+  it('should render KnativeServiceResources on sidebar', () => {
+    const svcItem: OverviewItem = {
```
Comment (vikram-raj) [inline frontend/packages/knative-plugin/src/components/overview/__tests__/OverviewDetailsKnativeResourcesTab.spec.tsx]: Define the props type using ```React.ComponentProps```. Avoid importing the prop type in spec file.

Comment (invincibleJai) [reply frontend/packages/knative-plugin/src/components/overview/__tests__/OverviewDetailsKnativeResourcesTab.spec.tsx]: Thanks :)

File: frontend/packages/knative-plugin/src/components/overview/__tests__/OverviewDetailsKnativeResourcesTab.spec.tsx [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+import EventSinkServicesOverviewList from '../EventSinkServicesOverviewList';
+
+type OverviewDetailsKnativeResourcesTabProps = React.ComponentProps<
+  typeof OverviewDetailsKnativeResourcesTab
+>;
+
+describe('OverviewDetailsKnativeResourcesTab', () => {
+  it('should render KnativeServiceResources on sidebar', () => {
+    const svcItem: OverviewDetailsKnativeResourcesTabProps = {
+      item: {
```
[the remark this answers is not in the document]
Comment (invincibleJai) [reply frontend/packages/knative-plugin/src/components/overview/__tests__/OverviewDetailsKnativeResourcesTab.spec.tsx]: Done

**Comment by andrewballantyne** [comment]
/hold cancel
/retest
#3847 has fixed the e2e tests

**Pull request merged**